test/Feature/HLSLLib/asuint16.int16.test

```diff
@@ -0,0 +1,98 @@
+#--- source.hlsl
+StructuredBuffer<uint16_t4> In0 : register(t0);
+StructuredBuffer<int16_t4> In1 : register(t1);
+
+RWStructuredBuffer<uint16_t4> Out0 : register(u2);
+RWStructuredBuffer<uint16_t4> Out1 : register(u3);
+
+
+[numthreads(1,1,1)]
+void main() {
+
+Out0[0] = asuint16(In0[0]);
+Out0[1] = uint16_t4(asuint16(In0[1].xyz), asuint16(In0[1].w));
+Out0[2] = uint16_t4(asuint16(In0[2].xy), asuint16(In0[2].zw));
+Out0[3] = asuint16(uint16_t4(0, 100, 65535, 10));
+  
+Out1[0] = asuint16(In1[0]);
+Out1[1] = uint16_t4(asuint16(In1[1].xyz), asuint16(In1[1].w));
+Out1[2] = uint16_t4(asuint16(In1[2].xy), asuint16(In1[2].zw));
+Out1[3] = asuint16(int16_t4(0, 32767, -100, -32768));
+}
+//--- pipeline.yaml
+
+---
+Shaders:
+  - Stage: Compute
+    Entry: main
+    DispatchSize: [1, 1, 1]
+Buffers:
+  - Name: In0
+    Format: UInt16
+    Stride: 8
+    Data: [0, 100, 65535, 10, 20, 30, 40, 50, 60, 70, 80, 6000]
+  - Name: In1
+    Format: Int16
+    Stride: 8
+    Data: [0, 32767, -100, -32768, 10, 20, 30, 6000, -3000, -30000, 50, 60]
+  - Name: Out0
+    Format: UInt16
+    Stride: 8
+    ZeroInitSize: 32
+  - Name: ExpectedOut0
+    Format: UInt16
+    Stride: 8
+    Data: [0, 100, 65535, 10, 20, 30, 40, 50, 60, 70, 80, 6000, 0, 100, 65535, 10]
+  - Name: Out1
+    Format: Int16
+    Stride: 8
+    ZeroInitSize: 32
+  - Name: ExpectedOut1
+    Format: Int16
+    Stride: 8
+    Data: [ 0, 32767, -100, -32768, 10, 20, 30, 6000, -3000, -30000, 50, 60, 0, 32767, -100, -32768 ]
```

Comment:
I don't understand why there are negative values, shouldn't asuint interpret the bits as an unsigned integer, and thus eliminate the sign in the final representation?